### DISCUSSION TITLE
CLI Starter Templates/Boilerplate for Chakra UI with Next.js (Pages Router and App Router)

### DISCUSSION THREAD

**Discussion by thesohailjafri** [chakra-ui/chakra-ui#9308] [Ideas]
### Description

Requesting CLI-based starter Templates/Boilerplate for Next.js projects using Chakra UI, designed to work seamlessly with both the Pages Router and App Router. This would allow developers to quickly set up projects by running a single command, similar to existing examples like Cloudinary's Next.js starter.

### Problem Statement/Justification

While Chakra UI provides excellent GitHub examples for Pages Router and App Router setups, a CLI-based starter template would simplify and streamline the setup process. Many developers prefer using a CLI command like `create-next-app` or [nextjs-bolierplate](https://vercel.com/templates/next.js/nextjs-boilerplate) to bootstrap projects quickly and efficiently without manually cloning repositories or setting up configurations.

Having an official CLI starter template would:

- Save time for developers.
- Ensure a consistent setup process for Chakra UI projects.
- Reduce friction for newcomers adopting Chakra UI with Next.js.
- Increase adoption of best practices with pre-configured examples.

### Proposed Solution or API

Provide starter templates that can be initialized directly through the create-next-app CLI command. For example:

```bash
npx create-next-app --example chakra-pages-router chakra-v3-pages-router-app
npx create-next-app --example chakra-app-router chakra-v3-app-router-app
```
Each template could include:

- Pre-configured ChakraProvider and theme setup.
- Examples of commonly used Chakra UI components (e.g., buttons, forms, modals).
- Folder structure optimized for either the Pages Router or App Router.
- Setup for Chakra UI's dark mode and responsive design features.
- Tailored integration with popular Next.js features like server-side rendering (SSR) and static site generation (SSG).
- Additionally, provide deployment instructions or integration with platforms like Vercel to make the process end-to-end.

### Alternatives

- Relying on the current GitHub examples, which require manual cloning and setup.
- Third-party unofficial templates that may lack updates or community support.
- Manually configuring Chakra UI for every project, which can be repetitive and error-prone.

### Additional Information

Resources to assist in this effort:

- Next.js CLI examples for inspiration on CLI-based [templates](https://vercel.com/templates/next.js).
- Chakra UI's existing GitHub examples for Pages and App Router.

This addition would be a natural evolution of Chakra UI's ecosystem, making it even easier for developers to build stunning Next.js applications with Chakra UI from the ground up.

**Comment by segunadebayo** [comment]
No official `create-next-app --example` template.

Use https://chakra-ui.com/docs/get-started/frameworks/next-app (or the pages-router guide). Snippet add is optional.
